### ISSUE TITLE
How to add other LV2 plugins to the ISO/USB stick

### ISSUE THREAD

**Issue by rominator1983** [moddevices/mod-live-usb#12]
First of all. Thanks for this thing. I might be building something around it with a notebook and a USB audio interface for rehearsals and stuff.

What I would like to do is add some (LV2) plugins that are available in guitarix (ubuntu studio) to the ISO. But I do not know how to go around about this and if this is even possible (considering plugin-GUI). If you could head me in the right direction this would be great. If time allows I could make a PR to adapt the documentation on how to achieve this.

I think for normal users it would be best to place things on the USB stick in some shared space/folder to be used/copied after startup and not include additional plugins in the build process of the ISO.

**Comment by falkTX** [comment]
All plugins come from mod-plugin-builder, with packages defined in https://github.com/moddevices/mod-live-usb/blob/main/plugins/Makefile
Regular desktop plugins are NOT supported, intentionally, because they do not behave as intended in the MOD platform (mainly due to lack of modgui)

**Comment by rominator1983** [comment]
thanks for the quick response. appreciate it. 

so let me rephrase what you said. If I wanted additional plugins in mod-live-usb I would have to add them to mod-plugin-builder and make sure they do work (Make, compilation and of course GUI and all) and then add them to the Makefile you linked me too?

I suppose I could also make PRs for mod-plugin-builder to add plugins?

I am thinking mostly about additional amp sims like this: https://github.com/GuitarML/NeuralPi

I will check out if that's something I want to do.

**Comment by falkTX** [comment]
Yeah, first add the plugins to MOD plugin builder, then to this repo.

**Comment by rominator1983** [comment]
thx. i'll give this a go

**Comment by rominator1983** [comment]
While we are at it. How would you say could one best bring his impulse responses and stuff to the ISO or the stick so they can be used conveniently later on?
When the ISO is written to USB and mounted before reboot things aren't obvious to me.

Also thinking about bringing GxRat to mod-builder as I think those guys have a point: https://forum.moddevices.com/t/emulation-rat-pedal/8049/6
So basically I would have to bring this https://github.com/brummer10/guitarix/tree/master/trunk/src/LV2/gx_aclipper.lv2 to lv2-data/plugins-fixed of mod-plugin-builder as it is done for example with the gx_muff (coming from https://github.com/brummer10/guitarix/tree/master/trunk/src/LV2/gx_muff.lv2) and provide the missing stuff. Right?

I read that @brummer10 did port pedals for the mod?

thx in advance.
btw. your quick replies are reaaaally appreciated!

**Comment by falkTX** [comment]
> btw. your quick replies are reaaaally appreciated!

dont take them for granted ;)

for any extra files on the ISO, I began to do live-usb persistence, but it does not always work.
the idea is to take a very specific offset on the usb raw data, and create a ext4 partition there. with a little safety checks for usb vs read-only (like dvd or mounted iso).
I think that is the best option here. see https://github.com/moddevices/mod-live-usb/blob/main/archiso/liveusb/airootfs/usr/local/bin/mod-live-rwdata for the script. (maybe it already works for you? try saving some pedalboards and rebooting)
